File: DAL/Dtos/MeterDto.cs
namespace DAL.Dtos
{
    public class MeterDto
    {
        public int Id { get; set; }

        public string? MeterName { get; set; }

        public string? MeterNumber { get; set; }

        public string? DocPath { get; set; }

        public DateTime PrevCheckDate { get; set; }

        public DateTime NextCheckDate { get; set; }

        public int ApartmentId { get; set; }

        public ApartmentDto? Apartment { get; set; }

        public MeterLocationDto? MeterLocation { get; set; }
    }
}

File: Utilities/ClientApp/src/app/core/Models/MeterDto.ts
import { ApartmentDto } from "./ApartmentDto";
import { MeterLocationDto } from "./MeterLocationDto";

export interface MeterDto {
    id?: number;
    meterName: string;
    meterNumber: string;
    docPath: string;
    prevCheckDate: Date;
    nextCheckDate: Date;
    apartmentId: number;
    apartment: ApartmentDto;
    meterLocation: MeterLocationDto;
}